saleor/graphql/discount/tests/test_voucher_channel_listing.py

```diff
@@ -0,0 +1,577 @@
+    voucher = voucher_without_channel
+    voucher_id = graphene.Node.to_global_id("Voucher", voucher.pk)
+    channel_id = graphene.Node.to_global_id("Channel", channel_USD.id)
+    variables = {
+        "id": voucher_id,
+        "input": {"addChannels": [{"channelId": channel_id, "discountValue": 5.5}]},
+    }
+
+    # when
+    response = api_client.post_graphql(
+        VOUCHER_CHANNEL_LISTING_UPDATE_MUTATION, variables=variables,
+    )
+
+    # then
+    assert_no_permission(response)
+
+
+def test_voucher_channel_listing_create_many_channel(
+    staff_api_client,
+    voucher_without_channel,
+    permission_manage_discounts,
+    channel_USD,
+    channel_PLN,
+):
+    # given
+    voucher = voucher_without_channel
+    voucher_id = graphene.Node.to_global_id("Voucher", voucher.pk)
+    channel_id = graphene.Node.to_global_id("Channel", channel_USD.id)
+    channel_pln_id = graphene.Node.to_global_id("Channel", channel_PLN.id)
+    variables = {
+        "id": voucher_id,
+        "input": {
+            "addChannels": [
+                {
+                    "channelId": channel_id,
+                    "discountValue": 5.5,
+                    "minAmountSpent": 10.1,
+                },
+                {
+                    "channelId": channel_pln_id,
+                    "discountValue": 50.5,
+                    "minAmountSpent": 100.2,
+                },
+            ]
+        },
+    }
+
+    # when
+    response = staff_api_client.post_graphql(
+        VOUCHER_CHANNEL_LISTING_UPDATE_MUTATION,
+        variables=variables,
+        permissions=(permission_manage_discounts,),
+    )
+    content = get_graphql_content(response)
+
+    # then
+    data = content["data"]["voucherChannelListingUpdate"]["voucher"]
+    assert not content["data"]["voucherChannelListingUpdate"]["discountErrors"]
+    channel_listing = data["channelListing"]
+    assert channel_listing[0]["channel"]["slug"] == channel_USD.slug
+    assert channel_listing[0]["discountValue"] == 5.5
+    assert channel_listing[0]["minSpent"]["amount"] == 10.1
+    assert channel_listing[0]["currency"] == channel_USD.currency_code
+    assert channel_listing[1]["channel"]["slug"] == channel_PLN.slug
+    assert channel_listing[1]["discountValue"] == 50.5
+    assert channel_listing[1]["minSpent"]["amount"] == 100.2
+    assert channel_listing[1]["currency"] == channel_PLN.currency_code
+
+
+def test_voucher_channel_listing_create_and_remove(
+    staff_api_client, voucher, permission_manage_discounts, channel_USD, channel_PLN,
+):
+    # given
+    voucher_id = graphene.Node.to_global_id("Voucher", voucher.pk)
+    channel_id = graphene.Node.to_global_id("Channel", channel_USD.id)
+    channel_pln_id = graphene.Node.to_global_id("Channel", channel_PLN.id)
+    variables = {
+        "id": voucher_id,
+        "input": {
+            "addChannels": [
+                {
+                    "channelId": channel_pln_id,
+                    "discountValue": 50.5,
```

Comment:
Small improvement, if you are reusing this value, it will be better to store this value in variable.